app/src/main/java/com/example/kuba10/mypokemonplaces/AddPlaceFragment/AddPlaceFragment.java

```diff
@@ -1,169 +0,0 @@
-import android.widget.TextView;
-
-import com.example.kuba10.mypokemonplaces.ChooseFragment.ChooseFragment;
-import com.example.kuba10.mypokemonplaces.Constants;
-import com.example.kuba10.mypokemonplaces.FragmentListener;
-import com.example.kuba10.mypokemonplaces.Model.PokePlace;
-import com.example.kuba10.mypokemonplaces.Model.PokemonGo;
-import com.example.kuba10.mypokemonplaces.R;
-import com.google.android.gms.maps.model.LatLng;
-import com.squareup.picasso.Picasso;
-
-import java.util.ArrayList;
-import java.util.Date;
-
-import butterknife.BindView;
-import butterknife.ButterKnife;
-
-
-
-public class AddPlaceFragment extends Fragment {
-
-    private FragmentListener fragmentListener;
-    private ArrayList<PokemonGo> pokemonGo_data_list;
-    private PokemonGo selectedPokemon;
-
-    @BindView(R.id.pokemon_image_view)
-    ImageView pokemonImageView;
-    @BindView(R.id.description_input)
-    TextView descriptionField;
-    @BindView(R.id.title_input)
-    TextView titleField;
-    @BindView(R.id.add_button)
-    Button addButt;
-    @BindView(R.id.cancel_button)
-    Button cancelButt;
-
-    public static AddPlaceFragment newInstance(LatLng position) {
-        AddPlaceFragment fragment = new AddPlaceFragment();
-        Bundle args = new Bundle();
-        args.putParcelable(Constants.POSITION, position);
-        fragment.setArguments(args);
-        return fragment;
-    }
-
-
-    @Override
-    public void onAttach(Context context) {
-        super.onAttach(context);
-        fragmentListener = (FragmentListener) context;
-        pokemonGo_data_list = fragmentListener.getPokemonList();
-    }
-    @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
-        View view = inflater.inflate(R.layout.fragment_add_place, container, false);
-
-        ButterKnife.bind(this, view);
-
-        addButt.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-
-                LatLng position = getArguments().getParcelable(Constants.POSITION);
-                double lat = position.latitude;
-                double lng = position.longitude;
-
-                PokePlace place = new PokePlace();
-
-                if (titleField.getText().toString().equals("")) {
-                    fragmentListener.showSnackbar(getString(R.string.placeMustHaveTitle));
-                } else {
-
-                    place.setTitle(titleField.getText().toString());
-                    place.setTitle(titleField.getText().toString());
-                    place.setDesctription(descriptionField.getText().toString());
-                    place.setLong(lng);
-                    place.setLat(lat);
-                    place.setListPosition(Constants.INDEX_NOT_SET);
-                    place.setFavourite(0);
-
-                    setSelectedPokemonId(place);
-                    setPlaceId(place);
```

Comment:
84-93 linijki wyekstraktuj do metody typu `inflatePlace` albo coś takiego. Nie wolisz tego zrobić w konstruktorze? Jeśli nie, to zastosuj wzorzec buildera.

Reply:
wybrałem opcje z builderem - zawsze dobrze przećwiczyć - ale nie do końca wiem czemu musiał być statyczną klasą - coś zrobiłem źle czy tak może być?

Reply:
O wiele lepiej gdy klasa wewnętrzna jest klasą statyczną - chodzi o to, że klasa wewnętrzna nie jest wtedy uzależniona od klasy zewnętrznej w kontekście pamięcie i Garbage Colectora. Wszystko gra :guitar: